lib/index.js

```diff
@@ -469,13 +500,33 @@ NodeHttpXHR.prototype.send = function (data) {
     this._resp = resp;
     this._responseText = '';
 
-    resp.setEncoding('utf8');
+    // var contentType = resp.headers['content-type'];
+    // TODO: adjust responseType from content-type header if applicable
+
+    // from Node API docs: The encoding argument is optional and only applies when chunk is a string. Defaults to 'utf8'.
+    if (this._responseType === 'text' || this._responseType === '') {
+      resp.setEncoding('utf8');
+    }
+
     resp.on('data', function onData(chunk) {
-      this._responseText += chunk;
+
+      if (typeof chunk === 'string') {
+        this._responseText += chunk;
+      } else if (typeof chunk === 'object') {
+        if (!(chunk instanceof Buffer)) {
```

Comment:
i am not even sure we need this check, was more of a dev-time tool

Reply:
I suppose it's good to have the assertion there, for sanity's sake. Maybe we want to have a catch-all, since there are only 2 possibilities (if I'm reading it correctly)?

```javascript
} else if (chunk instanceof Buffer) {
  // as before
} else {
  throw new Error('Assertion failed: Response-data should be either a string or Buffer');
}
```

Reply:
that's a good one. i am actually always recommending run-time assertions that slam into devs face when they break stuff, or even at runtime, instead of hiding away unexpected behavior :)